runtime-modules/project-token/src/tests/fixtures.rs

```diff
@@ -0,0 +1,865 @@
+                Payment::<Balance> {
+                    remark: "remark".as_bytes().to_vec(),
+                    amount: DEFAULT_SPLIT_PARTICIPATION,
+                },
+            )]
+            .into_iter()
+            .collect(),
+        );
+        Self {
+            sender: DEFAULT_ACCOUNT_ID.into(),
+            token_id: 1u64.into(),
+            outputs,
+        }
+    }
+
+    pub fn execute_call(&self) -> DispatchResult {
+        let state_pre = sp_io::storage::root();
+        let result = Token::transfer(
+            Origin::signed(self.sender),
+            self.token_id,
+            self.outputs.clone(),
+        );
+        let state_post = sp_io::storage::root();
+
+        // no-op in case of error
+        if result.is_err() {
+            assert_eq!(state_pre, state_post)
+        }
+
+        result
+    }
+}
+
+pub struct ClaimRevenueSplitAmountFixture {
+    sender: AccountId,
+    token_id: TokenId,
+}
+
+impl ClaimRevenueSplitAmountFixture {
+    pub fn default() -> Self {
+        Self {
+            sender: OTHER_ACCOUNT_ID.into(),
+            token_id: TokenId::one(),
+        }
+    }
+
+    pub fn with_account(self, account_id: u64) -> Self {
+        Self {
+            sender: account_id.into(),
+            ..self
+        }
+    }
+
+    pub fn with_token_id(self, token_id: u64) -> Self {
+        Self {
+            token_id: token_id.into(),
+            ..self
+        }
+    }
+
+    pub fn execute_call(&self) -> DispatchResult {
+        let state_pre = sp_io::storage::root();
+        let result = Token::claim_revenue_split_amount(Origin::signed(self.sender), self.token_id);
+        let state_post = sp_io::storage::root();
+
+        // no-op in case of error
+        if result.is_err() {
+            assert_eq!(state_pre, state_post)
+        }
+
+        result
+    }
+}
+
+pub struct AbandonRevenueSplitFixture {
+    sender: AccountId,
+    token_id: TokenId,
+}
+
+impl AbandonRevenueSplitFixture {
+    pub fn default() -> Self {
+        Self {
+            sender: OTHER_ACCOUNT_ID.into(),
+            token_id: TokenId::one(),
+        }
+    }
+
+    pub fn with_account(self, account_id: u64) -> Self {
+        Self {
+            sender: account_id.into(),
+            ..self
+        }
+    }
+
+    pub fn with_token_id(self, token_id: u64) -> Self {
+        Self {
+            token_id: token_id.into(),
+            ..self
+        }
+    }
+
+    pub fn execute_call(&self) -> DispatchResult {
+        let state_pre = sp_io::storage::root();
+        let result = Token::abandon_revenue_split(Origin::signed(self.sender), self.token_id);
+        let state_post = sp_io::storage::root();
+
+        // no-op in case of error
+        if result.is_err() {
+            assert_eq!(state_pre, state_post)
+        }
+
+        result
+    }
+}
+
+pub fn treasury_account_for(token_id: u64) -> AccountId {
+    TokenModuleId::get().into_sub_account(token_id)
+}
```

Comment:
This will always return the same account, but that's something specific for tests, because `ModuleId` is 8 bytes and `u64` (which is mock `AccountId`) is also 8 bytes.

`ModuleId.into_sub_account(token_id)` basically just takes the account prefix value (4 bytes), joins it with `ModuleId` and `TokenId` (in that case) and converts back to `AccountId`, but since prefix + `ModuleId` are already > 8 bytes, the `TokenId` value is just ignored.

In production, the `AccountId` is 32 bytes, so `prefix + ModuleId + TokenId` will convert to different `AccountId` for different `TokenId` and our current test environment is not reflecting this properly.

Reply:
I added all the required functionalities starting from: 723e8b634a3442d6bd08ac9bf666b9245889ead8